includes/payment-methods/class-upe-split-payment-gateway.php

```diff
@@ -377,6 +377,20 @@ public function create_token_from_setup_intent( $setup_intent_id, $user ) {
 		}
 	}
 
+	/**
+	 * Mandate must be shown and acknowledged by customer before deferred intent UPE payment can be processed.
+	 * This applies to SEPA and Link payment methods.
+	 * https://stripe.com/docs/payments/finalize-payments-on-the-server
+	 *
+	 * @return boolean True if mandate must be shown and acknowledged by customer before deferred intent UPE payment can be processed, false otherwise.
+	 */
+	public function does_payment_method_require_mandate_data() {
+		$is_link_payment       = Payment_Method::CARD === $this->get_selected_stripe_payment_type_id() && in_array( Payment_Method::LINK, $this->get_upe_enabled_payment_method_ids(), true );
```

Comment:
Is it accurate to say that `$is_link_payment` could be `true` if the payment method was CC and the merchant enabled Stripe Link? (i.e.: does this check accurately tell us if the customer paid with Stripe Link, and not just with CC?)
And would it matter?

Reply:
> Is it accurate to say that $is_link_payment could be true if the payment method was CC and the merchant enabled Stripe Link?

Yes. But it'd be inaccurate to say that `$is_link_payment` could be `true` **if and only if** the payment method was CC and the merchant chose Stripe Link. Because for the use-case, when someone's paying with CC w/o choosing Stripe Link (but having it enabled), `$is_link_payment` will be `true` as well.

> does this check accurately tell us if the customer paid with Stripe Link, and not just with CC?

Unfortunately, no. 

The check will be `true` for Stripe Link enabled, even if it's not used. Consequently, the `mandate_data` will be sent for plain CC payments as well. [This logic](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payment-gateway-wcpay.php#L1540C3-L1545) is being followed to support the process. I think that the Stripe Link initialisation logic on the back-end has a room for improvement and should be further improved/refined.

> And would it matter?

`mandate_data` is added to the payment intent request only in case we're confirming the intent, and Stripe uses it to validate that we've shown the terms to a user before the order placement. I think Stripe takes this parameter only if it's needed (e.g. when SEPA payment is used) and checks if we've sent it. For card payments, this parameter will be redundant and in my opinion won't be checked by Stripe. 

I think I need to change `$is_link_payment` name to something like `$is_link_enabled` to better reflect the reality. (done in 361cef7f534a4e9768715a61ed8fdbb49635448c) Other than that, I'm happy to refine this existing Stripe Link logic further if needed. Let me know your thoughts!

Reply:
Thank you for the explanation! I like the rename you've done, I think it'll provide better clarity for future wanderers and adventurers 👍